src/main/java/org/waarp/common/database/DbAdmin.java

```diff
@@ -76,262 +63,262 @@
      * different DbSession) is allowed by JDBC.
-     * 
+     *
      * @param model
      * @param server
      * @param user
-     * @param password
+     * @param passwd
      * @param write
      * @throws WaarpDatabaseSqlException
      * @throws WaarpDatabaseNoConnectionException
      */
-    public DbAdmin(DbModel model, String server, String user, String password,
-            boolean write) throws WaarpDatabaseNoConnectionException {
+    public DbAdmin(DbModel model, String server, String user, String passwd,
+                   boolean write) throws WaarpDatabaseNoConnectionException {
         this.server = server;
         this.user = user;
-        this.password = password;
+        this.passwd = passwd;
         this.dbModel = model;
         this.typeDriver = model.getDbType();
         if (typeDriver == null) {
             logger.error("Cannot find TypeDriver");
             throw new WaarpDatabaseNoConnectionException(
                     "Cannot find database driver");
         }
-
-        ds = new BasicDataSource();
-        try {
-            ds.setDriverClassName(DriverManager.getDriver(server).getClass().getName());
-        } catch (SQLException e) {
-            throw new WaarpDatabaseNoConnectionException(
-                    "Cannot find database driver");
+        if (write) {
+            for (int i = 0; i < RETRYNB; i++) {
+                try {
+                    setSession(new DbSession(this, false));
+                } catch (WaarpDatabaseNoConnectionException e) {
+                    logger.warn("Attempt of connection in error: " + i, e);
+                    continue;
+                }
+                isReadOnly = false;
+                getSession().setAdmin(this);
+                validConnection();
+                getSession().useConnection(); // default since this is the top
+                // connection
+                return;
+            }
+        } else {
+            for (int i = 0; i < RETRYNB; i++) {
+                try {
+                    setSession(new DbSession(this, true));
+                } catch (WaarpDatabaseNoConnectionException e) {
+                    logger.warn("Attempt of connection in error: " + i, e);
+                    continue;
+                }
+                isReadOnly = true;
+                getSession().setAdmin(this);
+                validConnection();
+                getSession().useConnection(); // default since this is the top
+                // connection
+                return;
+            }
         }
-        ds.setUrl(this.server);
-        ds.setUsername(this.user);
-        ds.setPassword(this.password);
-
-        ds.setDefaultAutoCommit(true);
-        ds.setDefaultReadOnly(!write);
-        ds.setValidationQuery(this.dbModel.getValidationQuery());
-        readOnly = !write;
-
-        validateConnection();
-        this.session = new DbSession(this, readOnly); 
+        setSession(null);
+        setActive(false);
```

Comment:
All calls of `setActive()` in this class should be removed, as it is deprecated...